spec/features/users_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :feature do
+  let!(:users) { create_list(:user, 5) }
+
+  describe 'Users index page' do
+    before { visit users_path }
+
+    it 'displays usernames of all users' do
+      users.each do |user|
+        expect(page).to have_content(user.name)
+      end
+    end
+
+    it 'displays profile pictures of all users' do
+      users.each do |user|
+        expect(page).to have_selector("img[src$='#{user.photo}']")
+      end
+    end
+
+    it 'displays the number of posts each user has written' do
+      users.each do |user|
+        expect(page).to have_content("Number of posts: #{user.posts.count}")
+      end
+    end
+
+    it 'redirects to user show page when a user is clicked' do
+      first_user = users.first
+      click_link first_user.name
+      expect(current_path).to eq(user_path(first_user))
+    end
+  end
+
+  describe 'User show page' do
+    let(:user) { create(:user) }
+    before { visit user_path(user) }
+
+    it 'displays the user profile picture' do
+      expect(page).to have_selector("img[src$='#{user.photo}']")
+    end
+
+    it 'displays the user username' do
+      expect(page).to have_content(user.name)
+    end
+
+    it 'displays the number of posts the user has written' do
+      expect(page).to have_content("Number of posts: #{user.posts.count}")
+    end
+
+    it 'displays the user bio' do
+      expect(page).to have_content(user.bio)
+    end
+
+    it 'displays the user first 3 posts' do
+      user.recent_posts.each do |post|
+        expect(page).to have_content(post.text.truncate(100))
+      end
+    end
+
+    it 'has a button to view all of the user posts' do
+      expect(page).to have_link('See all posts', href: user_posts_path(user))
+    end
+
+    it 'redirects to post show page when a post is clicked' do
+      post = user.posts.first
+      click_link "Post ##{post.id}"
+      expect(current_path).to eq(user_post_path(user, post))
+    end
```

Comment:
- [ ] Nice writing a test to checking the redirects to the post's show page when you click on a user's post. :clap: However this test has failed(see image). Kindly check it, and make sure all the test are passing
![Screenshot from 2023-10-10 17-09-41](https://github.com/bohaz/BlogApp/assets/91142432/5522c966-b6fb-48b2-9f18-928c41381711)